### ISSUE TITLE
Firmware compatibility with chinese devices

### ISSUE THREAD

**Issue by Harvie** [nanovna-v2/NanoVNA-V2-firmware#68]
Hello,
is this firmware compatible with following device?

https://www.aliexpress.com/item/-/1005001413753569.html

I suspect that is not THE original NanoVNA, but i really like the steel box providing good shielding and very good mechanical protection for the connectors. Also i live in Czech Republic and the free shipping option is rather important for me, because shipping from US usualy costs more than the device itself. On the other hand i want to acknowledge the original NanoVNA developers, but my budget is rather limited. I've heard about official NanoVNA store on Aliexpress, but it seem to be gone.

Is the provided device a good option in terms of opensource firmware support?

**Comment by gabriel-tenma-white** [comment]
If you don't need >1GHz then a V1 nanovna is around 30usd on aliexpress, for example [this](https://www.aliexpress.com/item/4001159371329.html) or [this](https://www.aliexpress.com/item/1005001548276258.html).
The one you linked is supported by the V2_2 firmware, but all clones include cables that are no good above 2GHz so you will want to buy better quality cables on mouser: https://www.mouser.com/ProductDetail/Johnson-Cinch-Connectivity-Solutions/415-0033-018/

**Comment by Harvie** [comment]
> The one you linked is supported by the V2_2 firmware, 

What are the downfalls when compared to V2_plus and V2_plus4? Will V2_2 be supported in the future firmware releases?

> If you don't need >1GHz then a V1 nanovna is around 30usd on aliexpress

Are there any benefits from buying V1 instead of the device i've linked? I don't need it to be that cheap. I would happily pay bit more than $54 for device to get firmware upgrades in the future. But i like the mechanical robustnes of metal case. And while i don't really need >1GHz right now, but i think it would be interresting to be able to do basic comparison of Wi-Fi antennas. So 3GHz sounds reasonable.

> you will want to buy better quality cables on mouser

Yes i plan to do that anyway, because i need different connectors.

**Comment by gabriel-tenma-white** [comment]
V2_2 firmware is still kept updated for now but some things like averaging work better on plus and plus4 because of the sweep speed. V2_2 with metal enclosure has slightly worse isolation, and besides Plus4 it seems only -F V2 has solved this.

It really depends what you expect from it. If all you need to do is measure antennas then the above is a non-issue, and the 10dB return loss cables that come with V2 clones will work just fine since it is calibrated out anyway. But if you do need cables with better than 20dB return loss and cal kit load with better than 35dB return loss and the best dynamic range, you will probably end up spending more with a clone since you have to buy your own cables and cal kit.

**Comment by Harvie** [comment]
> V2_2 with metal enclosure has slightly worse isolation

What does that mean? Is that crosstalk between the two ports? I wonder if that might be calibrated out in firmware. kinda like echo cancelation works in cellphones.

**Comment by gabriel-tenma-white** [comment]
Yes crosstalk. It is calibrated out but drifts after some time. If you recalibrate every time you use the VNA it doesn't matter.
